### PR TITLE
Modify lit_autoupdate so that it can handle errors in prelude.carbon

### PR DESCRIPTION
**Pull request by jonmeow**
The intent here is that changes to prelude.carbon shouldn't break every test that expects some error from prelude.carbon; that would be too fragile. As a consequence, this effectively ignores the line number in prelude.carbon.

This is a little complex because we don't know which line in the original source file is actually causing the error, just that there is an error. Also, the previous look-behind approach required a fixed-with prefix, whereas we want a little more than that in order to capture the filename for comparison.

This would be hard to do with extra_check_replacement because the path to bazel.runfiles is complex to calculate. As a consequence, this is basically all new code.

### CODE REVIEW

File: bazel/testing/lit_autoupdate_base.py [hunk 1]
```diff
@@ -190,35 +203,61 @@ class CheckLine(Line):
...
         self.indent = ""
         self.out_line = out_line.rstrip()
-        self.line_number_format = line_number_format
+        self.line_number_delta_prefix = line_number_delta_prefix
         self.line_number_pattern = line_number_pattern
-        self.line_numbers = [
-            int(n) - 1 for n in line_number_pattern.findall(self.out_line)
-        ]
+
+        # If any match has a match specific to this file, use the first line for
```
Comment (zygoloid) [inline bazel/testing/lit_autoupdate_base.py]: Should this be:
```suggestion
        # If any match is specific to this file, use the first line for
```
? I'm not sure what it means for a match to have a match. Or maybe
```suggestion
        # If any match refers to a line in this file, use the first line for
```
I also misinterpreted "first line" as meaning "the first line in the file", not "the first line in the match", maybe "first such line" would be clearer?

File: bazel/testing/lit_autoupdate_base.py [hunk 2]
```diff
@@ -283,6 +323,9 @@ def get_matchable_test_output(
             (test, f"{{{{.*}}}}/{test}"),
         ],
     )
+    out = bazel_runfiles.sub("{{.*}}/", out)
+    # Replacing runfiles is a more complex replacement.
+    # We have some things show up under runfiles; this removes them.
```
Comment (zygoloid) [inline bazel/testing/lit_autoupdate_base.py]:
```suggestion
    # Replacing runfiles is a more complex replacement.
    # We have some things show up under runfiles; this removes them.
    out = bazel_runfiles.sub("{{.*}}/", out)
```
I think these comments are referring to the code before them, not the code after?

File: explorer/testdata/assert/fail_assert.carbon [hunk 3]
```diff
@@ -6,7 +6,7 @@
 // so that it doesn't need to be updated on every prelude change.
 //
 // This can't be autoupdated because the error line comes form a different file.
```
Comment (zygoloid) [inline explorer/testdata/assert/fail_assert.carbon]: Please remove these outdated comments too.

**Review by zygoloid — APPROVED** [review]
LGTM with some minor comments.

File: explorer/testdata/assert/fail_assert.carbon (hunk 3 above)
Comment (jonmeow) [reply explorer/testdata/assert/fail_assert.carbon]: Done

File: explorer/testdata/operators/fail_left_shift_large_rhs.carbon [hunk 4]
```diff
@@ -6,7 +6,7 @@
 // so that it doesn't need to be updated on every prelude change.
 //
 // This can't be autoupdated because the error line comes form a different file.
-// NOAUTOUPDATE
+// AUTOUPDATE
```
Comment (chandlerc) [inline explorer/testdata/operators/fail_left_shift_large_rhs.carbon:9]: Several files seem to still have the no-auto-update explanation.

**Review by chandlerc — COMMENTED** [review]
FYI

**Pull request merged**